keep-ui/app/alerts/alert-table.tsx

```diff
@@ -217,14 +227,74 @@ export function AlertTable({
     }),
   ];
 
+  const extraPayloadKeys = Array.from(
+    new Set(
+      alerts
+        .map((alert) => {
+          const extraPayload = getExtraPayloadNoKnownKeys(alert);
+          return Object.keys(extraPayload.extraPayload);
+        })
+        .reduce((acc, keys) => [...acc, ...keys], [])
+    )
+  );
+  // Create all the necessary columns
+  const extraPayloadColumns = extraPayloadKeys.map((key) =>
+    columnHelper.display({
+      id: key,
+      header: key,
+      cell: (context) => {
+        const val = (context.row.original as any)[key];
+        if (typeof val === "object") {
+          return JSON.stringify(val);
+        }
+        return (context.row.original as any)[key]?.toString() ?? "";
+      },
+    })
+  );
+  const columnsToHideFromLocalStorage = localStorage.getItem(
+    getHiddenColumnsLocalStorageKey(presetName)
+  );
+  const columnOrderLocalStorage = localStorage.getItem(
+    getColumnsOrderLocalStorageKey(presetName)
+  );
+
+  const [columns] = useState<typeof defaultColumns>(() => [
+    ...defaultColumns,
+    ...extraPayloadColumns,
+  ]);
```

Comment:
Is there a reason why `columns` has to be a state variable? I don't see it being manipulated anywhere

Reply:
Not really, I kinda followed an example from react-table.